core/src/main/kotlin/io/specmatic/core/pattern/JSONObjectPattern.kt

```diff
@@ -27,6 +27,19 @@ data class JSONObjectPattern(
     val minProperties: Int? = null,
     val maxProperties: Int? = null
 ) : Pattern {
+
+    override fun eliminateOptionalKey(value: Value, resolver: Resolver): Value {
+        if (value !is JSONObjectValue) return value
+
+        val filteredPattern = this.pattern.filterKeys { !it.endsWith("?") }
```

Comment:
filteredPattern and filteredJsonObject variable names should be renamed for clarity. Perhaps mandatoryObjectPatternMap and mandatoryObjectMap.

Reply:
Thanks, I've fixed that